vignettes/examples/hierarchical_linear_general.Rmd

```diff
@@ -0,0 +1,31 @@
+### Hierarchical linear regression in general conditional formulation
+
+A hierarchical, Bayesian linear regression model using the iris data, with random intercepts and slopes for each of the three species.
+The slopes and intercepts for each species are *correlated* in this example. We allow every species to have a species specific slope for `Sepal.Length`.
+
+
+```{r hierarchical_linear_general_greta}
+int  <- normal(0, 10)
+coef <- normal(0, 10)
+sd   <- cauchy(0, 3, truncation = c(0, Inf))
+
+n.species  <- length(unique(iris$Species))
+species_id <- as.numeric(iris$Species)
+
+Z     <- matrix(0, nrow(iris), n.species * 2)
+gamma <- zeros(n.species * 2)
+
+for (s in unique(species_id)) {
+  Z[species_id == s, s * 2  - 1] <- 1
+  Z[species_id == s, s * 2]  <- iris$Sepal.Length[species_id == s]
+  ranef_sd <- diag(2)
+  gamma[(s * 2 - 1):(s * 2)] <- multivariate_normal(rep(0, 2), ranef_sd) 
```

Comment:
as it's currently set up, this is the same (the prior has the same density) as:
```r
gamma <- normal(0, 1, dim = n.species * 2)
```
That might be simpler for people to understand, if that's what you wanted to go for. The coefficients will still be correlated (in the posterior), though it you wanted to explicitly model the correlations it would require a variable for the variance-covariance matrix `ranef_sd`

Reply:
Yeah, I am aware. I had an inverse Wishart for the covariance, but that throws an TF error (see below). However, I find this formulation still more valuable, since it shows explicitely how the gammas, should be constructed. Once I figure out, why the Wishart does not work here, I'd have changed it back.